### ISSUE TITLE
TBooleanMap does not free allocated memory in Destroy

### ISSUE THREAD

**Issue by SAASTN** [graphics32/graphics32#325]
It appears that the `TBooleanMap` class does not free the allocated memory in its destructor. Instead, it only sets `FBits` to `nil`. 

Is there a specific reason for this implementation? 

As I encountered a memory leak report the first time I used `TBooleanMap`, I am concerned that this type might not have been widely used or tested. Is it safe to use it?

**Comment by andersmelander** [comment]
The bug was introduced when the class was refactored to use dynamic memory instead of a dynamic array.
I'll fix it when I get back from vacation.

**Comment by andersmelander** [comment]
I have a few hours before take-off so I fixed it now.

> I am concerned that this type might not have been widely used or tested. Is it safe to use it?

It has probably not seen much general use but it's an old class so I think it's safe to use. I will write a unit test for it later.

The only place `TBooleanMap` is used internally in Graphics32 is in the `TContourRasterizer` class which in turn is used in the **Mandelbrot** and **NestedSampling** examples.
I just made some quick tests with the Mandelbrot example to verify that it works and after fixing a bunch of unrelated bugs in the example I can conclude that it now works and doesn't leak. It also works fine in the NestedSampling example.
